diff --git a/packages/apps/Browser/src/com/android/browser/BrowserPreferencesPage.java b/packages/apps/Browser/src/com/android/browser/BrowserPreferencesPage.java
index 9b538e0..319dcbe 100644
--- a/packages/apps/Browser/src/com/android/browser/BrowserPreferencesPage.java
+++ b/packages/apps/Browser/src/com/android/browser/BrowserPreferencesPage.java
@@ -106,8 +106,10 @@ public class BrowserPreferencesPage extends PreferenceActivity {
             "com.android.browser.preferences.PrivacySecurityPreferencesFragment",
             "com.android.browser.preferences.AccessibilityPreferencesFragment",
             "com.android.browser.preferences.AdvancedPreferencesFragment",
+            "com.android.browser.preferences.WebsiteSettingsFragment",
             "com.android.browser.preferences.BandwidthPreferencesFragment",
-            "com.android.browser.preferences.LabPreferencesFragment"));
+            "com.android.browser.preferences.LabPreferencesFragment",
+            "com.android.browser.preferences.DebugPreferencesFragment"));

     @Override
     protected boolean isValidFragment(String fragmentName) {
diff --git a/packages/apps/Browser/src/com/android/browser/ComboViewActivity.java b/packages/apps/Browser/src/com/android/browser/ComboViewActivity.java
index 7053c9b..29b40ae 100644
--- a/packages/apps/Browser/src/com/android/browser/ComboViewActivity.java
+++ b/packages/apps/Browser/src/com/android/browser/ComboViewActivity.java
@@ -71,6 +71,8 @@ public class ComboViewActivity extends Activity implements CombinedBookmarksCall
         mTabsAdapter = new TabsAdapter(this, mViewPager);
         mTabsAdapter.addTab(bar.newTab().setText(R.string.tab_bookmarks),
                 BrowserBookmarksPage.class, args);
+        mTabsAdapter.addTab(bar.newTab().setText(R.string.tab_history),
+                BrowserHistoryPage.class, args);

         if (savedInstanceState != null) {
             bar.setSelectedNavigationItem(


